contracts/misc/LensPeripheryDataProvider.sol

```diff
@@ -14,7 +14,7 @@ import {Errors} from '../libraries/Errors.sol';
  *
  * @dev This is useful because it allows clients to filter out follow NFTs that were transferred to
  * a recipient by another user (i.e. Not a mint) and not register them as "following" unless
- * the recipient explicitly toggles the follow here. 
+ * the recipient explicitly toggles the follow here.
  */
 contract LensPeripheryDataProvider {
```

Comment:
Why does the dataprovider have functions that are not view or pure? As a data provider, I would expect it to only provide data and not be able to alter any state or emit events. This could be another helper, but do not seem like it should be this one. Also, if I burn my follower nft or transfer it, would the data provider still not let you think that I am a follower?

Reply:
So you're right, this isn't really a "data provider" per se. Technically speaking it's just a peripheral helper contract, with some convenient functions that don't *really* fit anywhere else. 

As for the follow toggling, since this just emits an event, it acts as a canonical source of truth for when you receive a follow NFT from another user. Upon burning or transferring it, since the appropriate events are emitted, it's expected that this previously emitted event will be rendered obsolete in the timeline.

Reply:
Renamed to `LensPeriphery` in 1e9901de23ba51ddccd0a100176c40402d4a113d!

Reply:
Fixed in your PR and merged https://github.com/aave/lens-protocol/pull/84!